### ISSUE TITLE
Ignore warning Variable 'A.B.C' is used, but not defined

### ISSUE THREAD

**Issue by p74** [nokia/RED#268]
I am currently moving from ride to red and a face some issues when validating rf files
I have a c;ass that looks like

class A
   class B
       C = "foo"

in rf i get the class A
  ${A} = getClass

I got the error from red: Variable 'A.B.C' is used, but not defined
but when i run rf everything is fine. I try to ignore this error but as far as I can see it is not possible.
How can this error be ignored without turning off the whole validation?

**Comment by michalanglart** [comment]
Hi,

1. What version of RED do you use?
2. Could you please wrap the code into formatted paragraphs?

It looks a bit similar to this one: #242 although I'm not sure how to read your code because of formatting.

**Comment by p74** [comment]
I have update the orginal post

**Comment by michalanglart** [comment]
Ok looks better now although I've started looking into that but there are more questions coming as your example is insufficient for me.

I understand you have a single `.py` file with the code similar to the one in your first comment (similar because this one has syntax errors due to missing `:` characters in classes definitions). I don't know however what is your Robot file doing with this file. Is it imported as a library and then you obtain instance of `A` library class? Is it imported as variables file? In which place is this error reported? The same file or some other?

Would it be possible to provide some minimal project which shows this validation marker in place you don't expect it?

**Comment by p74** [comment]
I did some more try's
i think its not related to the nested class but this error is caused by
setting a variable in my toplevel suite as global variable with keyword "Set Global Variable" and use this variable in another suite

because when i add the code below to the suite the validation is going ok. 
 ```
*** Variables ***
${PLATFORM}    ${PLATFORM}
```

**Comment by p74** [comment]
its difficult to share my actual code but i will try to make a working example for you

**Comment by michalanglart** [comment]
Yes, RED will report that the variable is not recognized if it is set by `Set Test/Suite/Global Variable` keyword and used in some different test or suite. This is discussed in couple of issues - see #7, #68, #195 

There are some workarounds for this in above-mentioned issues

**Comment by p74** [comment]
ok thx for your feed back i will try to find a work a round

**Comment by michalanglart** [comment]
Ok, so my comment above applies here. The issue #195 is exactly the same setup as here. Please familiarize yourself with the issues I've mentioned above.

When RED validates Robot files it performs static analysis but the variables created using `Set Test/Suite/Global Variable` keywords are created dynamically and in general we do not know if they will be created or not. I've expanded on this topic more deeply here: https://github.com/nokia/RED/issues/195#issuecomment-375589938
